compiler/lib/reserved.ml

```diff
@@ -140,7 +140,24 @@ let provided =
     ; "isNaN"
     ; "parseFloat"
     ; "parseInt"
-    ; "module"
-    ; "require"
+    ; "module" (* only available in node *)
+    ; "require" (* only available in node *)
     ; "Symbol"
+    ; "ArrayBuffer"
+    ; "Float32Array"
+    ; "Float64Array"
+    ; "Int16Array"
+    ; "Int32Array"
+    ; "Int8Array"
+    ; "TextDecoder"
+    ; "Uint16Array"
+    ; "Uint32Array"
+    ; "Uint8Array"
+    ; "atob"
+    ; "clearInterval"
+    ; "console"
+    ; "global" (* only available in node *)
```

Comment:
Do we really need this ? One can now use the portable `globalthis` to refer to the global object.

Reply:
In node, `global === globalThis`, so either can be used. Since `globalThis` wasn't added to node until v10, a lot of older JS that exists would still have `global` being used.

I added this because Binaryen compiled to JS has both `globalThis` and `global` for its feature detection logic.